src/core/surface/call.c

```diff
@@ -425,7 +425,13 @@ static grpc_cq_completion *allocate_completion(grpc_call *call) {
     if (call->allocated_completions & (1u << i)) {
       continue;
     }
-    call->allocated_completions |= (gpr_uint8)(1u << i);
+    /* NB: the following integer arithmetic operation needs to be in its
+     * expanded form due to the "integral promotion" performed (see section
+     * 3.2.1.1 of the C89 draft standard), which in this case upcasts the result
+     * of the bitwise OR to "unsigned". A cast to the smaller container type is
```

Comment:
The 1u introduces unsigned. The gpr_uint8 cast typecasts it back down to unsigned char. Then it has to get integer promoted before it can be operated on by bitwise or, which is why it turns back to int. goo.gl/ZLboLg .